src/clipboard_handlers/tables_clipboard.ts

```diff
@@ -1,110 +1,181 @@
-  ) {
-    if (!("tables" in content) || !target.zones.length) {
+  private getTableStyleToCopy(cellPosition: CellPosition): TableStyle {
+    const styleFromTable = this.getters.getCellTableStyle(cellPosition);
+    const cellStyle = this.getters.getCellStyle(cellPosition);
+
+    const bordersFromTable = this.getters.getCellTableBorder(cellPosition);
+    const cellBorder = this.getters.getCellBorder(cellPosition);
+
+    return {
+      style: { ...styleFromTable, ...removeFalsyAttributes(cellStyle) },
+      border: { ...bordersFromTable, ...removeFalsyAttributes(cellBorder) },
+    };
+  }
+
+  paste(target: ClipboardPasteTarget, content: ClipboardContent, options?: ClipboardOptions) {
+    if (!content || !content.tableCells) {
       return;
     }
     const zones = target.zones;
     const sheetId = this.getters.getActiveSheetId();
     if (!options?.isCutOperation) {
-      if (options?.pasteOption === undefined) {
-        this.pasteCopiedTables(sheetId, zones, content);
-      }
+      this.pasteFromCopy(sheetId, zones, content.tableCells, options);
     } else {
-      this.pasteFromCut(sheetId, zones, content);
+      this.pasteFromCut(sheetId, zones, content, options);
     }
   }
 
-  private pasteFromCut(sheetId: UID, target: Zone[], content: ClipboardContent) {
-    for (const table of content.tables) {
-      this.dispatch("REMOVE_TABLE", {
-        sheetId: content.sheetId,
-        target: [table.zone],
-      });
+  private pasteFromCut(
+    sheetId: UID,
+    target: Zone[],
+    content: ClipboardContent,
+    options?: ClipboardOptions
+  ) {
+    for (const row of content.tableCells) {
+      for (const tableCell of row) {
+        if (tableCell.table) {
+          this.dispatch("REMOVE_TABLE", {
+            sheetId: content.sheetId,
+            target: [tableCell.table.zone],
+          });
+        }
+      }
     }
-    this.pasteCopiedTables(sheetId, target, content);
+    const selection = target[0];
+    this.pasteZone(sheetId, selection.left, selection.top, content.tableCells, options);
   }
 
-  /** Paste the tables that are in the state */
-  private pasteCopiedTables(sheetId: UID, target: Zone[], content: ClipboardContent) {
-    const selection = target[0];
-    const cutZone = content.zones[0];
-    const cutOffset: [number, number] = [
-      selection.left - cutZone.left,
-      selection.top - cutZone.top,
-    ];
-    for (const table of content.tables) {
-      const newTableZone = createAdaptedZone(table.zone, "both", "MOVE", cutOffset);
-      this.dispatch("CREATE_TABLE", { sheetId, target: [newTableZone], config: table.config });
-      for (const i of range(0, table.filtersValues.length)) {
+  protected pasteZone(
+    sheetId: UID,
+    col: HeaderIndex,
+    row: HeaderIndex,
+    tableCells: TableCell[][],
+    clipboardOptions?: ClipboardOptions
+  ) {
+    for (const [r, rowCells] of tableCells.entries()) {
+      for (const [c, tableCell] of rowCells.entries()) {
```

Comment:
`entries()` allocates new arrays. It's not be the best if there are lot's of them.